src/tests/receiving/receivingStatusChangesWithoutPartialReceiving.test.ts

```diff
@@ -0,0 +1,575 @@
+import AppConfig from '@/config/AppConfig';
+import { ShipmentType } from '@/constants/ShipmentType';
+import { expect, test } from '@/fixtures/fixtures';
+import InboundListPage from '@/pages/inbound/list/InboundListPage';
+import StockMovementShowPage from '@/pages/stockMovementShow/StockMovementShowPage';
+import { StockMovementResponse } from '@/types';
+import { getToday } from '@/utils/DateUtils';
+
+test.describe('Status changes on sm view page when receive shipment in location without partial receiving', () => {
+  let STOCK_MOVEMENT: StockMovementResponse;
+  const description = 'some description';
+  const dateRequested = getToday();
+
+  test.beforeEach(
+    async ({
+      supplierLocationService,
+      stockMovementService,
+      mainProductService,
+      otherProductService,
+      depotLocationService,
+    }) => {
+      const supplierLocation = await supplierLocationService.getLocation();
+      const depotLocation = await depotLocationService.getLocation();
+      const PRODUCT_ONE = await mainProductService.getProduct();
+      const PRODUCT_TWO = await otherProductService.getProduct();
+
+      STOCK_MOVEMENT = await stockMovementService.createInbound({
+        originId: supplierLocation.id,
+        destinationId: depotLocation.id,
+        description,
+        dateRequested,
+      });
+
+      await stockMovementService.addItemsToInboundStockMovement(
+        STOCK_MOVEMENT.id,
+        [
+          { productId: PRODUCT_ONE.id, quantity: 20 },
+          { productId: PRODUCT_TWO.id, quantity: 10 },
+        ]
+      );
+
+      await stockMovementService.sendInboundStockMovement(STOCK_MOVEMENT.id, {
+        shipmentType: ShipmentType.AIR,
+      });
+    }
+  );
+
+  test.afterEach(async ({ stockMovementShowPage, authService }) => {
+    await stockMovementShowPage.goToPage(STOCK_MOVEMENT.id);
+    const isRollbackLastReceiptButtonVisible =
+      await stockMovementShowPage.rollbackLastReceiptButton.isVisible();
+    const isRollbackButtonVisible =
+      await stockMovementShowPage.rollbackButton.isVisible();
+
+    // due to failed test, shipment might not be received which will not show the button
+    if (isRollbackLastReceiptButtonVisible) {
+      await stockMovementShowPage.rollbackLastReceiptButton.click();
+    }
+
+    if (isRollbackButtonVisible) {
+      await stockMovementShowPage.rollbackButton.click();
+    }
+
+    await stockMovementShowPage.clickDeleteShipment();
+    await authService.changeLocation(AppConfig.instance.locations.main.id);
+  });
+
+  test('Assert status changes on view page and receipt tab when receive 1 item partially', async ({
+    stockMovementShowPage,
+    receivingPage,
+    authService,
+  }) => {
+    await test.step('Go to stock movement show page', async () => {
+      await authService.changeLocation(AppConfig.instance.locations.depot.id);
```

Comment:
Why this is required here? what is this achieving? If this is part of each, should it be a part of before each?